[Fleet] Update README and local setup dev docs (#184629)
elastic/kibana: x-pack/plugins/fleet/dev_docs/local_setup/enrolling_agents.md

    description: Fleet server policy
    namespace: default
    package_policies:
      - name: Fleet Server
        package:
          name: fleet_server
        inputs:
          - type: fleet-server
            keep_enabled: true
            vars:
              - name: host
                value: 0.0.0.0
                frozen: true
              - name: port
                value: 8220
                frozen: true

# Set a default Fleet Server host.
xpack.fleet.fleetServerHosts:
  - id: default-fleet-server
    name: Default Fleet server
    is_default: true
    # host_urls: [https://<the-IP-address-of-your-VM>:8220] # For running a Fleet Server in a VM
    # host_urls: ['https://host.docker.internal:8220'] # For running a Fleet Server Docker container

# Set a default Elasticsearch output.
xpack.fleet.outputs:
  - id: es-default-output
    name: Default output
    type: elasticsearch
    is_default: true
    is_default_monitoring: true
    # hosts: ['http://<your-local-IP>:9200'] # For enrolling agents on VM
    # hosts: ['http://host.docker.internal:9200'] # For enrolling dockerized agents
```

## Using Multipass VMs

> [Multipass](https://multipass.run/) is lightweight virtualization tool for running Ubuntu VMs. Follow instructions at https://multipass.run/install or to install Multipass on your local machine.

Reviewer: ```suggestion
[Multipass](https://multipass.run/) is a lightweight virtualization tool for running Ubuntu VMs. Follow the instructions at https://multipass.run/install to install Multipass on your local machine.
```

Advantages of running Elastic Agents on a VM include:
- More realistic setup.
- Ability to use the `elastic-agent` commands, e.g. `sudo elastic-agent status`, `sudo elastic-agent restart`...
- Agents can be upgraded.
- Elastic Defend can be installed.

> To run a Fleet Server and agents on VMs, make sure the default output host defined in your `kibana.dev.yml` uses your local IP (NB: using `localhost` can cause connection issues). The default Fleet Server host can be set once the VM for the Fleet Server is running. In Fleet UI, this should be reflected in Fleet settings:

Reviewer 2: quite silly but can you include how to retrieve the correct local IP to use?

Reviewer 3: We can probably provide the Mac way which is 

```
$ ipconfig getifaddr en0
```

Author: No such thing as silly, this is the kind of stuff that can save someone time 🙂 I usually Alt+click on my wifi dropdown menu in the Mac top bar, but +1 to the command line way 👍

![Fleet settings UI showing hosts for agents on VM](./screenshots/vm_fleet_settings.png)

### Running a Fleet Server

1\. Launch a Multipass instance for your Fleet Server:
```sh
multipass launch --name fleet-server --disk 10G --network en0
```

Available options are detailed at https://multipass.run/docs/launch-command.

It is generally recommended to provide additional disk space (default 5G) for running Elastic Agents.

In addition, the `--network` option adds a network interface to the instance, in this case `en0`. This allows the Fleet Server instance to communicate with the enrolled agents via the wifi network interface. You can find out the IP address by running:
```sh
multipass list
```

Example output:
```sh
Name                    State             IPv4             Image
fleet-server            Running           192.168.1.1     Ubuntu 24.04 LTS
                                          192.168.1.100
```

Copy the second IP address into the host URLs of the Fleet Server host in your `kibana.dev.yml`. Wait for Kibana to restart.

2\. Shell into the instance:
```sh
multipass shell fleet-server
```

> 3\. Open Kibana in the browser and head over to Fleet. initially, there should be no Fleet Server:

Reviewer: ```suggestion
3\. Open Kibana in a browser and head over to Fleet. Initially, there should be no Fleet Server:
```

5\. Run the `sudo ./elastic-agent install` command provided in the instructions, adding the `--insecure` flag as we are connecting to Elasticsearch using HTTP. After a few seconds the UI should confirm that the agent is enrolled and shipping data:
![Fleet UI showing add agent on VM flow, step 2](./screenshots/vm_add_agent_2.png)

### Gotchas

1\. The system clock within Multipass instances stops when the host computer is suspended (see https://askubuntu.com/questions/1486977/repeated-incorrect-time-in-multipass-clients-with-ubuntu-22-04). This can result in a running Elastic Agent being incorrectly "in the past" after your laptop was asleep for a while. The easiest fix is to restart all Multipass instances, which will reset their clocks:
```sh
multipass restart --all
```

2\. As noted in the enrollment steps above, the architecture of a Multipass VM matches the one of the host. If the agent installation fails, check that the downloaded agent binary has the correct architecture type.

## Using Docker containers

Official documentation: https://www.elastic.co/guide/en/fleet/current/elastic-agent-container.html

> The main advantage of running Elastic Agents on a Docker container is a one command setup that can be easily to scripted (see [Using the `run_dockerized_agent.sh` script](#using-the-run_dockerized_agentsh-script) below).There are a few limitations, however, including:

Reviewer: ```suggestion
The main advantage of running Elastic Agents in a Docker container is a one command setup that can easily be scripted (see [Using the `run_dockerized_agent.sh` script](#using-the-run_dockerized_agentsh-script) below). There are a few limitations, however, including:
```
